src/pages/AdminTemplates/AdminTemplates.tsx

```diff
@@ -0,0 +1,154 @@
+              className="templates-page__nav-buttons--cancel"
+              onClick={() => setSelectedTemplate(undefined)}
+            >
+              Back to the templates
+            </button>
+            <button
+              className="templates-page__nav-buttons--confirm"
+              onClick={() => console.log("asd")}
+            >
+              Choose this template
+            </button>
+          </div>
+          <div className="templates-page__title">{selectedTemplate.title}</div>
+          <div className="templates-page__subtitle">
+            {selectedTemplate.description}
+          </div>
+          <div className="templates-page__recipes">
+            {recipes.map((recipe, index) => (
+              <button
+                key={`${recipe.title}-${index}`}
+                onClick={() => setSelectedRecipe(recipe)}
+                className={classNames("templates-page__recipe", {
+                  "templates-page__recipe--selected": recipe === selectedRecipe,
+                })}
+              >
+                {recipe.title}
+              </button>
+            ))}
+          </div>
+          <div className="templates-page__template-preview">
+            <PartyMapContainer
+              rooms={[]}
+              iconsMap={{}}
+              coordinatesBoundary={{ width: 60, height: 60 }}
+              interactive={false}
+              resizable={false}
+              backgroundImage={selectedTemplate.background}
+            />
+          </div>
+        </>
+      )}
+      {!selectedTemplate && (
+        <>
+          <div className="templates-page__title">Choose a Template</div>
+          <div className="templates-page__subtitle">
+            You can edit it during the next steps
+          </div>
+          <button
+            disabled={isLoading}
+            className={classNames("templates-page__header-button", {
+              "templates-page__header-button--disabled": isLoading,
+            })}
+          >{`I'd rather start from scratch`}</button>
+          <div className="templates-page__templates-section">
+            {templates.map((template, index) => (
+              <div
+                key={`${template.title}-${index}`}
+                className={classNames("templates-page__template", {
+                  "templates-page__template--disabled": isLoading,
+                })}
+                onClick={() => selectTemplate(template)}
+              >
+                <div
+                  style={{ backgroundImage: `url(${template.background})` }}
+                  className="templates-page__template-background"
+                />
+                <div className="templates-page__template-title">
+                  {template.title}
+                </div>
+                <div className="templates-page__template-description">
+                  {template.description}
+                </div>
+              </div>
+            ))}
+          </div>
+        </>
+      )}
```

Comment:
This should be extracted out into a sub-component and/or refactored. There's a lot of duplication in structure within this block and the block above it. We shouldn't be doing that. Common structure should use the same component, and then for the 'slots' that need to differ, we can 'inject' that content as needed.

Reply:
They both seem to have `title`, `subtitle`, a button, then their 'main content' section.